content/en/blog/_posts/2019-10-20-csi-ephemeral-inline-volumes.md

```diff
@@ -0,0 +1,75 @@
+---
+title: CSI Ephemeral Inline Volumes
+date: 2019-10-20
+---
+
+**Author:** Patrick Ohly (Intel)
+
+Typically, volumes provided by an external storage driver in
+Kubernetes are *persistent*, with a lifecycle that is completely
+independent of pods or (as a special case) loosely coupled to the
+first pod which uses a volume ([late binding
+mode](https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode)).
+The mechanism for requesting and defining such volumes in Kubernetes
+are [Persistent Volume Claim (PVC) and Persistent Volume
+(PV)](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)
+objects. Originally, volumes that are backed by a Container Storage Interface
+(CSI) driver could only be used via this PVC/PV mechanism.
+
+But there are also use cases for data volumes whose content and
+lifecycle is tied to a pod. For example, a driver might populate a
+volume with dynamically created secrets that are specific to the
+application running in the pod. Such volumes need to be created
+together with a pod and can be deleted as part of pod termination
+(*ephemeral*). They get defined as part of the pod spec (*inline*).
```

Comment:
Might be good to give an example, like: https://github.com/kubernetes-csi/csi-driver-image-populator#example-usage-in-kubernetes

Reply:
I was hoping that readers will patient enough to continue reading and will then find real examples in the currently missing sections for PMEM-CSI, image populator and cert manager.

Reply:
That works too.